src/Models/ProductAttributeOption.php

```diff
@@ -0,0 +1,129 @@
+            });
+
+            $builder->leftJoin('eav_attribute_option_value as store_value', function ($join) {
+                $join->on('eav_attribute_option.option_id', '=', 'store_value.option_id')
+                    ->where('store_value.store_id', static::getCurrentStoreId());
+            });
+
+            $builder->select([
+                'eav_attribute_option.option_id',
+                'eav_attribute_option.attribute_id',
+                'eav_attribute_option.sort_order',
+                'eav_attribute.attribute_code',
+                'eav_attribute.frontend_label as attribute_label',
+                'admin_value.value as admin_value',
+                'store_value.value as store_value',
+            ]);
+        });
+    }
+
+    public function attribute()
+    {
+        return $this->belongsTo(ProductAttribute::class, 'attribute_id', 'attribute_id');
+    }
+
+    public function AdminValue(): Attribute
+    {
+        return Attribute::make(
+            get: fn () => $this->attributes['admin_value'] ?? ''
+        );
+    }
+
+    public function StoreValue(): Attribute
+    {
+        return Attribute::make(
+            get: fn () => $this->attributes['store_value'] ?? $this->admin_value
+        );
+    }
+
+    public function DisplayValue(): Attribute
+    {
+        return Attribute::make(
+            get: fn () => $this->store_value ?: $this->admin_value
+        );
+    }
+
+    public function getCacheKey(): string
+    {
+        return "product_attribute_option_{$this->option_id}_store_" . static::getCurrentStoreId();
+    }
+
+    public function scopeRunwaySearch(Builder $query, string $search)
+    {
+        return $query->where(function ($q) use ($search) {
+            $q->where('eav_attribute_option.option_id', 'LIKE', "%{$search}%")
+                ->orWhere('eav_attribute.attribute_code', 'LIKE', "%{$search}%")
+                ->orWhere('eav_attribute.frontend_label', 'LIKE', "%{$search}%")
+                ->orWhere('admin_value.value', 'LIKE', "%{$search}%")
+                ->orWhere('store_value.value', 'LIKE', "%{$search}%")
+                ->orWhere('eav_attribute_option.sort_order', 'LIKE', "%{$search}%");
+        });
+    }
+
+    public function scopeByAttribute($query, $attributeId)
+    {
+        return $query->where('eav_attribute_option.attribute_id', $attributeId);
+    }
+
+    public function scopeByAttributeCode($query, $attributeCode)
+    {
+        return $query->where('eav_attribute.attribute_code', $attributeCode);
+    }
+
+    public function scopeOrderBySort($query)
+    {
+        return $query->orderBy('eav_attribute_option.sort_order');
+    }
+
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::saved(function ($model) {
+            Cache::forget($model->getCacheKey());
+        });
+
+        static::deleted(function ($model) {
+            Cache::forget($model->getCacheKey());
+        });
```

Comment:
I think `ProductAttribute` and `ProductAttributeOption` can use the same observer :)